integration/querier_test.go

```diff
@@ -814,7 +815,13 @@ func TestQuerierWithBlocksStorageOnMissingBlocksFromStorage(t *testing.T) {
 	_, err = c.Query("series_1", series1Timestamp)
 	require.Error(t, err)
 	assert.Contains(t, err.Error(), "500")
-	assert.Contains(t, err.(*promv1.Error).Detail, "consistency check failed because some blocks were not queried")
+
+	var promErr *promv1.Error
+	if errors.As(err, &promErr) {
```

Comment:
I think you can do `require.ErrorAs` and avoid the `if` condition here.

Reply:
I can, but then the message wouldn't be checked, no?

Reply:
I meant this:

```go
	var promErr *promv1.Error
	require.ErrorAs(t, err, &promErr)
	assert.Contains(t, promErr.Detail, "consistency check failed because some blocks were not queried")
```